File: dowoo/src/crawl/parser/shubaFamily.ts
// 69shuba.com / twkan.com은 동일한 사이트 엔진을 쓰고 있어서
// <script>var bookinfo = {...}</script>에 제목/이전장/다음장이 그대로 들어있다.
// DOM 선택자보다 안정적이라 이걸 우선 사용한다.
export interface Bookinfo {
  articlename: string
  chaptername: string
  previewPage: string | null
  nextPage: string | null
}

function extractField(html: string, key: string): string | null {
  const regex = new RegExp(`${key}\\s*:\\s*(?:'([^']*)'|"([^"]*)")`)
  const match = html.match(regex)
  if (!match) return null
  return match[1] ?? match[2] ?? null
}

export function extractBookinfo(html: string): Bookinfo {
  return {
    articlename: extractField(html, 'articlename') ?? '',
    chaptername: extractField(html, 'chaptername') ?? '',
    previewPage: extractField(html, 'preview_page'),
    nextPage: extractField(html, 'next_page'),
  }
}

File: dowoo-python/app/crawl/parsers/shuba_family.py
import re
from typing import Optional


def _extract_field(html: str, key: str) -> Optional[str]:
    pattern = re.compile(key + r"\s*:\s*(?:'([^']*)'|\"([^\"]*)\")")
    match = pattern.search(html)
    if not match:
        return None
    return match.group(1) if match.group(1) is not None else match.group(2)


def extract_bookinfo(html: str) -> dict:
    return {
        "articlename": _extract_field(html, "articlename") or "",
        "chaptername": _extract_field(html, "chaptername") or "",
        "previewPage": _extract_field(html, "preview_page"),
        "nextPage": _extract_field(html, "next_page"),
    }
